### PR TITLE
fix(STONEINTG-640): limit ITS name length

### PR DESCRIPTION
**Pull request by MartinBasti**
We rely on labels which are limited by 63 character, thus we have to limit IntegrationTestScenario name to the same limits as labels have.

## Maintainers will complete the following section

- [ ] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [ ] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [ ] [Controllers diagrams](https://github.com/redhat-appstudio/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

File: config/webhook/manifests.yaml [hunk 1]
```diff
@@ -1,32 +1,3 @@
-
----
-apiVersion: admissionregistration.k8s.io/v1
```
Comment (MartinBasti) [inline config/webhook/manifests.yaml:3]: This is automatically generated, and kubebuilder somehow decided it's not needed to have Mutating webhook.

I tested and conversion still works.

If you think that it should stay there let me know, but FYI I tried very hard to keep mutating webhook here and failed on endpoint errors :D

File: config/webhook/manifests.yaml [hunk 2]
```diff
@@ -52,6 +22,7 @@ webhooks:
     operations:
     - CREATE
     - UPDATE
+    - DELETE
     resources:
-    - applications
```
Comment (MartinBasti) [inline config/webhook/manifests.yaml:56]: Dunno why it was application before, I also don't know why validation webhook has been configured without any validation code.

File: api/v1beta1/integrationtestscenario_suite_test.go [hunk 3]
```diff
@@ -56,35 +63,67 @@ var _ = BeforeSuite(func() {
...
+	var err error
+	// cfg is defined in this file globally.
+	cfg, err = testEnv.Start()
 	Expect(err).NotTo(HaveOccurred())
 	Expect(cfg).NotTo(BeNil())
 
 	scheme := runtime.NewScheme()
-	Expect(AddToScheme(scheme)).To(Succeed())
+	err = AddToScheme(scheme)
+	Expect(err).NotTo(HaveOccurred())
```
Comment (dheerajodha) [inline api/v1beta1/integrationtestscenario_suite_test.go:81]: This change does the same thing but in 2 lines instead. Should we revert it?

[docs links](https://onsi.github.io/ginkgo/#suite-setup-and-cleanup-beforesuite-and-aftersuite)

-------------------
image

Comment (MartinBasti) [reply api/v1beta1/integrationtestscenario_suite_test.go:81]: I copied autogenerated code, that's why.

Comment (MartinBasti) [reply api/v1beta1/integrationtestscenario_suite_test.go]: Reverted to original

File: config/webhook/manifests.yaml (hunk 1 above)
Comment (dirgim) [reply config/webhook/manifests.yaml:3]: I think we should be fine with removing the mutating webhook, but @jsztuka please double check this.

**Review by dirgim — APPROVED** [review]
LGTM

File: api/v1beta1/integrationtestscenario_webhook.go [hunk 4]
```diff
@@ -17,11 +17,42 @@ limitations under the License.
...
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
 func (r *IntegrationTestScenario) SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(r).
 		Complete()
 }
+
+//+kubebuilder:webhook:path=/validate-appstudio-redhat-com-v1beta1-integrationtestscenario,mutating=false,failurePolicy=fail,sideEffects=None,groups=appstudio.redhat.com,resources=integrationtestscenarios,verbs=create;update;delete,versions=v1beta1,name=vintegrationtestscenario.kb.io,admissionReviewVersions=v1
```
Comment (jsztuka) [inline api/v1beta1/integrationtestscenario_webhook.go:33]: `name=vintegrationtestscenario.kb.io`
Was this generated by kubebuilder?

**Review by dheerajodha — APPROVED** [review]

File: config/webhook/manifests.yaml (hunk 1 above)
Comment (jsztuka) [reply config/webhook/manifests.yaml:3]: Mutating webhook should be capable of working properly even without manifests, wo yes, this should not interfere with its current behavior.

**Review by jsztuka — APPROVED** [review]
lgtm

**Pull request merged**